File: repository/src/main/java/info/mengnan/dialogerai/repository/entity/KnowledgeBase.java
package info.mengnan.dialogerai.repository.entity;

import com.baomidou.mybatisplus.annotation.FieldFill;
import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import info.mengnan.dialogerai.repository.enums.KnowledgeBaseStatus;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@TableName("knowledge_base")
public class KnowledgeBase {

    @TableId(type = IdType.AUTO)
    private Long id;

    private Long memberId;

    private String name;

    private String description;

    /** private / public */
    private String visibility;

    /** DRAFT：新建向导草稿；ACTIVE：正式知识库 */
    private KnowledgeBaseStatus status;

    /** ES 索引名：{memberId}_kb_{id} */
    private String indexName;

    /** 当前知识库构建任务 ID（async_task.task_id） */
    private String buildTaskId;

    /**
     * 返回最相似的 K 个文本分段
     */
    private Integer topK;

    /**
     * 只返回相似度分数高于此阈值的文本分段
     */
    private Double score;

    private Integer deleted;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;
}

File: ui/src/api/knowledgeBase.ts
import { $get, $post, $put, $delete } from './request'

export type KnowledgeBaseStatus = 'DRAFT' | 'ACTIVE'

export interface KnowledgeBaseItem {
  id: number
  memberId: number
  name: string
  description: string | null
  visibility: string
  status?: KnowledgeBaseStatus
  indexName: string
  documentCount: number
  buildTaskId?: string | null
  createdAt: string | null
  topK?: number | null
  score?: number | null
  creatorName?: string | null
}

export interface KnowledgeBaseCreatePayload {
  name: string
  description?: string
  visibility?: 'private' | 'public'
  topK?: number | null
  score?: number | null
}

export interface KnowledgeBaseCreateResult {
  id: number
  name: string
  indexName: string
  status?: KnowledgeBaseStatus
}

/** 创建知识库 */
export async function createKnowledgeBase(payload: KnowledgeBaseCreatePayload) {
  return $post<KnowledgeBaseCreateResult>('/kb', payload)
}

/** 知识库列表 */
export async function getKnowledgeBaseList() {
  return $get<KnowledgeBaseItem[]>('/kb/list')
}

/** 知识库详情 */
export async function getKnowledgeBase(kbId: number) {
  return $get<KnowledgeBaseItem>(`/kb/${kbId}`)
}

/** 更新草稿知识库基本信息 */
export async function updateKnowledgeBaseDraft(
  kbId: number,
  payload: KnowledgeBaseCreatePayload,
) {
  return $put<KnowledgeBaseItem>(`/kb/${kbId}`, payload)
}

/** 更新知识库基本信息（名称、描述、可见性） */
export async function updateKnowledgeBase(
  kbId: number,
  payload: KnowledgeBaseCreatePayload,
) {
  return $put<KnowledgeBaseItem>(`/kb/${kbId}`, payload)
}

/** 将草稿知识库设为正式（向导完成） */
export async function activateKnowledgeBase(kbId: number) {
  return $post<KnowledgeBaseItem>(`/kb/${kbId}/activate`)
}

/** 删除知识库（含草稿） */
export async function deleteKnowledgeBase(kbId: number) {
  return $delete(`/kb/${kbId}`)
}
